### PR TITLE
Run2-alca100 Update IsoTrack calibration system to deal with only L1 trigger (no HLT)

### PR DESCRIPTION
**Pull request by bsunanda**
Also takeout some unnecessary casting

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by lpernie** [comment]
+1

File: Calibration/HcalCalibAlgos/test/HcalIsoTrkAnalyzer.cc [hunk 1]
```diff
@@ -992,13 +996,12 @@ double HcalIsoTrkAnalyzer::trackP(const reco::Track* pTrack,
 
   double pmom = -1.0;
   if (genParticles.isValid()) {
-    reco::GenParticleCollection::const_iterator p;
     double mindR(999.9);
-    for (p=genParticles->begin(); p!=genParticles->end(); ++p) {
+    for (const auto &p : (reco::GenParticleCollection)(*genParticles)) {
```
Comment (davidlange6) [inline Calibration/HcalCalibAlgos/test/HcalIsoTrkAnalyzer.cc]: remove the (reco::GenParticleCollection)

File: Calibration/HcalCalibAlgos/test/HcalIsoTrkAnalyzer.cc [hunk 2]
```diff
@@ -1013,11 +1016,10 @@ double HcalIsoTrkAnalyzer::rhoh(const edm::Handle<CaloTowerCollection>& tower) {
   for (auto eta : etabins_) {
     for (auto phi : phibins_) {
       double hadder = 0;
-      for (CaloTowerCollection::const_iterator pf_it = tower->begin(); 
-	   pf_it != tower->end(); pf_it++) {
-	if (fabs(eta-pf_it->eta())>etahalfdist_)                 continue;
-	if (fabs(reco::deltaPhi(phi,pf_it->phi()))>phihalfdist_) continue;
-	hadder+=pf_it->hadEt();
+      for (const auto & pf_it : (CaloTowerCollection)(*tower)) {
```
Comment (davidlange6) [inline Calibration/HcalCalibAlgos/test/HcalIsoTrkAnalyzer.cc]: and the (CaloTowerCollection)

Comment (bsunanda) [reply Calibration/HcalCalibAlgos/test/HcalIsoTrkAnalyzer.cc]: done

File: Calibration/HcalCalibAlgos/test/HcalIsoTrkAnalyzer.cc (hunk 1 above)
Comment (bsunanda) [reply Calibration/HcalCalibAlgos/test/HcalIsoTrkAnalyzer.cc]: done

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@smuzaffar @davidlange6 @slava77 The failure is not due to this PR. Please advise

**Comment by smuzaffar** [comment]
please test
error was in IB which is fixed now

**Comment by arunhep** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**